File: app/src/test/java/com/example/androidfundamentals/di/MockAddressModule.java
package com.example.androidfundamentals.di;

import com.example.androidfundamentals.interactors.AddressInteractor;
import com.example.androidfundamentals.presenters.AddressPresenter;
import com.example.androidfundamentals.repositories.AddressRepository;
import com.google.gson.Gson;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

import static org.mockito.Mockito.mock;

@Module
public class MockAddressModule {

    @Provides
    @Singleton
    public AddressPresenter providesPresenter(AddressInteractor addressInteractor) {
        return mock(AddressPresenter.class);
    }

    @Provides
    public Gson providesGson()
    {
        return new Gson();
    }

    @Provides
    public AddressRepository providesAddressRepository()
    {
        return mock(AddressRepository.class);
    }

    @Provides
    public AddressInteractor providesAddressInteractor()
    {
        return mock(AddressInteractor.class);
    }
}

File: app/src/test/java/com/example/androidfundamentals/di/MockAddressModule.kt
package com.example.androidfundamentals.di

import com.example.androidfundamentals.interactors.AddressInteractor
import com.example.androidfundamentals.presenters.AddressPresenter
import com.example.androidfundamentals.repositories.AddressRepository

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import io.mockk.mockk

@Module
class MockAddressModule {

    @Provides
    @Singleton
    fun providesPresenter(): AddressPresenter {
        return mockk(relaxed = true)
    }

    @Provides
    fun providesAddressRepository(): AddressRepository {
        return mockk(relaxed = true)
    }

    @Provides
    @Singleton
    fun providesAddressInteractor(): AddressInteractor {
        return mockk(relaxed = true)
    }
}
